extensions/functions_geometry.yaml

```diff
@@ -0,0 +1,68 @@
+%YAML 1.2
+---
+scalar_functions:
+  -
+    name: "isvalid"
+    description: >
+      Return true if the input geometry is a valid 2D geometry.
+
+      For 3 dimensional and 4 dimensional geometries, the validity is still only tested
```

Comment:
If we're going to have this restriction it'd be nice to either have 2d in the name or an option that specifies the number of dimensions to check against.

Reply:
That's a good point.  I also noticed that even though postGIS sets this restriction (https://postgis.net/docs/ST_IsValid.html), some other backends don't specify the dimensionality.  For example: https://docs.snowflake.com/en/sql-reference/functions/st_isvalid

@paleolimbot, do you have any thoughts on this?